api/s2n.h

```diff
@@ -687,6 +687,269 @@ extern int s2n_cert_get_utf8_string_from_extension_data_length(const uint8_t *ex
+ * 
+ * Note that the identity is copied into s2n memory and the user is responsbile for freeing the memory 
+ * associated with the `identity` input. 
+ *
+ * @param psk A pointer to a PSK object to be updated with the identity.
+ * @param identity The identity in raw bytes format to be copied.
+ * @param identity_size The length of the PSK identity being set.
+ */
+S2N_API 
+int s2n_psk_set_identity(struct s2n_psk *psk, const uint8_t *identity, uint16_t identity_size);
+
+/**
+ * Sets the out-of-band/externally provisioned PSK secret for a given s2n_psk object.
+ *
+ * # Safety
+ *
+ * Note that the secret is copied into s2n memory and the user is responsbile for freeing the memory 
+ * associated with the `secret` input. 
+ *
+ * When using an out-of-band provisioned pre-shared secret, a critical consideration is using 
+ * sufficient entropy during the PSK generation. Deriving a shared secret from a password or 
+ * other low-entropy sources is not secure. A low-entropy secret, or password, is subject to 
+ * dictionary attacks based on the PSK binder. Refer: RFC8446 Section-2.2 for more information.
+ *
+ * @param psk A pointer to a PSK object to be updated with the secret.
+ * @param secret The secret in raw bytes format to be copied.
+ * @param secret_size The length of the pre-shared secret being set.
+ */
+S2N_API 
+int s2n_psk_set_secret(struct s2n_psk *psk, const uint8_t *secret, uint16_t secret_size);
+
+/**
+ * Sets the PSK HMAC algorithm for a given s2n_psk object. The supported PSK HMAC 
+ * algorithms are as listed in the enum `s2n_psk_hmac` above.
+ * 
+ * @param psk A pointer to the s2n_psk object to be updated with the PSK HMAC algorithm.
+ * @param hmac The PSK HMAC algorithm being set.  
+ */
+S2N_API 
+int s2n_psk_set_hmac(struct s2n_psk *psk, s2n_psk_hmac hmac);
+
+/**
+ * Appends a PSK object to the list of PSKs supported by the s2n connection. 
+ * If a PSK with a duplicate identity is found, an error is returned and the PSK is not added to the list.
+ * Note that the PSK object is copied to the connection and the user is responsbile for freeing the memory associated with the s2n_psk object. 
+ *
+ * @param conn A pointer to the s2n_connection object that contains the list of PSKs supported.
+ * @param psk A pointer to the `s2n_psk` object to be appended to the list of PSKs on the s2n connection.
+ */
+S2N_API 
+int s2n_connection_append_psk(struct s2n_connection *conn, struct s2n_psk *psk);
+
+/**
+ * The list of PSK modes supported by s2n-tls for TLS versions >= TLS1.3.
+ * Currently s2n-tls supports two modes - `S2N_PSK_MODE_RESUMPTION`, which represents the PSKs established 
+ * using the previous connection via session resumption, and `S2N_PSK_MODE_EXTERNAL`, which represents PSKs 
+ * established out-of-band/externally using a secure mutually agreed upon mechanism.
+ * Note that currently s2n-tls supports only PSK with ECDHE key exchange, which provides forward secrecy.
+ */ 
```

Comment:
Why is this note here? That's a different kind of "mode" unrelated to this enum. If we support more key exchange modes, they won't be added to this enum.